packages/backend/src/studio-api-impl.ts

```diff
@@ -1,27 +1,37 @@
 import type { StudioAPI } from '@shared/StudioAPI';
 import { Category } from '@shared/models/ICategory';
 import { Recipe } from '@shared/models/IRecipe';
-import content from './ai.json';
 import { AI_STUDIO_FOLDER, ApplicationManager } from './managers/applicationManager';
 import { RecipeStatusRegistry } from './registries/RecipeStatusRegistry';
 import { RecipeStatus } from '@shared/models/IRecipeStatus';
 import { ModelInfo } from '@shared/models/IModelInfo';
 import { TaskRegistry } from './registries/TaskRegistry';
 import { Task } from '@shared/models/ITask';
-import * as path from 'node:path';
 import { ModelResponse } from '@shared/models/IModelResponse';
 import { PlayGroundManager } from './playground';
 import * as podmanDesktopApi from '@podman-desktop/api';
 
+import * as path from 'node:path';
+import * as fs from 'node:fs';
+import { Catalog } from '@shared/models/ICatalog';
+
 export const RECENT_CATEGORY_ID = 'recent-category';
 
 export class StudioApiImpl implements StudioAPI {
+  
+  private catalog: Catalog;
+
   constructor(
     private applicationManager: ApplicationManager,
     private recipeStatusRegistry: RecipeStatusRegistry,
     private taskRegistry: TaskRegistry,
     private playgroundManager: PlayGroundManager,
-  ) {}
+  ) {
+    const catalogPath = path.resolve(__dirname, '..', 'catalog.json');
+    const catalogStr = fs.readFileSync(catalogPath, 'utf-8');
```

Comment:
I'm wondering here if you need to use a read of the file as it seems anyway you embed it into the extension

so could it like 
```
import catalog from '../catalog.json'
```

or with newer spec

```ts
import catalog from './catalog.json' with { type: 'json' };
```

so you don't need to add it in Containerfile, do not need to handle synchronous method/io like readFileSync instead of promises readFile, etc

Reply:
Probably, but I'll want in the next iteration to run a watch on this file, to detect changes, and I prefer to go one step in that direction

Reply:
note that if it's imported, vite will automatically reload the file as well and update the extension's code

Reply:
(so no need to setup a custom watcher for development mode)

Reply:
But what about the production code? I would like the extension to detect changes even when the app is running in production mode

Reply:
in the current code the file is embedded and read from the extension so it doesn't change the behaviour.

if you need to read from a custom folder, you'll still need to include a default file in case it's not yet provided by the user

and in production, anyway you should not write to the source directory and monitor the source directory. You'll need to copy/read the file from a different folder (extension storage folder) b/c here if you update the extension it'll override the file anyway.

so as you need to have an 'included' version anyway it's easier to import/include it as a resource of the extension rather than a separate file.

Reply:
ok, you are right, I didn't think about having an 'included' version. 
I'll proceed differently then.
Thanks

Reply:
but it's not a requested change on my side, etc.

Reply:
I changed the scenario to use the imported json file as default catalog if the user does not provide any catalog.